### DISCUSSION TITLE
Build instructions

### DISCUSSION THREAD

**Discussion by taylorh140** [hpcc-systems/hpcc-js-wasm#173] [Q&A] [answered]
I would like to build a wasm plugin for a different project and I'm curious how you were able to get graphviz 8.1.0 compiled to wasm. I would like to do something similar but I don't see any details for building that part. I was looking at some of the cmake files and the docker file but I don't think I see how it was done quite yet and I would really like to be able to duplicate your work.

If you could add some details on how to build the expat/graphviz dependencies components that would be really helpful.

**Comment by GordonSmith** [comment, marked as answer]
For the libs you mention I am using "vcpkg" to download and compile see:
1. https://github.com/hpcc-systems/hpcc-js-wasm/blob/trunk/vcpkg.json
2. https://github.com/hpcc-systems/hpcc-js-wasm/blob/trunk/scripts/cpp-install-vcpkg.sh

These create "static" libs for those libraries with emscripten.

Next those libe are linked in with the wrapper cpp file here:  https://github.com/hpcc-systems/hpcc-js-wasm/blob/trunk/src-cpp/graphviz/CMakeLists.txt

**Comment by taylorh140** [comment]
Thanks for this, I bet that took some time to figure out.

**Reply by taylorh140** [reply]
So, on my setup i installed vcpkg and when i clone the git directory and run.
~/./vcpkg/vcpkg install
I get a dissapointing:
error: while loading base91:
error: base91 does not exist

I think this is going to take some time to figure out.

**Reply by GordonSmith** [reply]
base91 is not in the official vcpkg ports list, so I added it as an "overlay" (see the vcpkg-overlays folder), the same is true for graphviz.
